app/lib/forms/qualified_teacher_check.rb

```diff
@@ -120,12 +113,33 @@ def active_alert?
       @active_alert
     end
 
+    def before_render
+      # Load info from current_user into store if it isn't already set
+      # This way this info will be pre-populated in the forms.
+      # This is useful for when the user is logged in and we already know this info.
+      wizard.store["full_name"] ||= query_store.current_user.full_name
+      wizard.store["trn"] ||= query_store.current_user.trn
+      wizard.store["date_of_birth"] ||= query_store.current_user.date_of_birth
+    end
+
     def after_save
       wizard.store["trn_verified"] = trn_verified?
       wizard.store["trn_lookup_status"] = trn_lookup_status
       wizard.store["trn_auto_verified"] = trn_auto_verified?
       wizard.store["verified_trn"] = verified_trn
       wizard.store["active_alert"] = active_alert?
+
+      query_store.current_user.update!(
+        trn: trn_to_store,
+        full_name:,
+        date_of_birth:,
+        national_insurance_number: ni_number_to_store,
+        trn_verified: trn_verified?,
+        trn_lookup_status:,
+        trn_auto_verified: !!trn_auto_verified?,
+        active_alert: active_alert?,
+      )
```

Comment:
Setting the info here immediately rather than at the end of the flow in handle_submission_for_store means that we can use a single place to find TRNs now, `current_user.trn`.